Patch-Source: https://github.com/fujita/tgt/pull/57
--
From 1a5e14e678ba944986428f4abc586a68e7cdf31a Mon Sep 17 00:00:00 2001
From: Alex Xu <351006+Hello71@users.noreply.github.com>
Date: Mon, 10 Jul 2023 20:21:37 +0000
Subject: [PATCH] fix posix_fallocate ret check

"posix_fallocate() returns zero on success, or an error number on failure. Note that errno is not set."
---
 usr/tgtimg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/tgtimg.c b/usr/tgtimg.c
index 99f91113..1f61b940 100644
--- a/usr/tgtimg.c
+++ b/usr/tgtimg.c
@@ -452,8 +452,8 @@ static int sbc_new(int op, char *path, char *capacity, char *media_type, int thi
 				exit(5);
 			}
 		} else {
-			if (posix_fallocate(fd, 0, size*1024*1024LL) == -1) {
-				perror("posix_fallocate failed.");
+			if (posix_fallocate(fd, 0, size*1024*1024LL) != 0) {
+				printf("posix_fallocate failed\n");
 				exit(3);
 			}
 		}
